newIDE/app/src/Leaderboard/useLeaderboardReplacer.js

```diff
@@ -0,0 +1,415 @@
+              <TableBody>
+                {erroredLeaderboards.map(({ leaderboardId, error }) => (
+                  <TableRow key={leaderboardId}>
+                    <TableRowColumn>{leaderboardId}</TableRowColumn>
+                    <TableRowColumn>{error.toString()}</TableRowColumn>
+                  </TableRow>
+                ))}
+              </TableBody>
+            </Table>
+          ) : null}
+        </ColumnStackLayout>
+      </Line>
+    </Dialog>
+  );
+};
+
+type RetryOrAbandonCallback = () => void;
+
+type UseLeaderboardReplacerOutput = {
+  /**
+   * Launch search through the whole project for leaderboard ids to replace.
+   */
+  findLeaderboardsToReplace: (project: gdProject, sourceGameId: string) => void,
+
+  /**
+   * Render, if needed, the dialog that will show the progress of leaderboard replacement.
+   */
+  renderLeaderboardReplacerDialog: () => React.Node,
+};
+
+type ErroredLeaderboard = {
+  leaderboardId: string,
+  error: Error,
+};
+
+/**
+ * Hook allowing to find and duplicate leaderboards in a project, useful after
+ * opening a project from an example.
+ */
+export const useLeaderboardReplacer = (): UseLeaderboardReplacerOutput => {
+  const [project, setProject] = React.useState<gdProject | null>(null);
+  const [gameId, setGameId] = React.useState<string | null>(null);
+  const [shouldReplace, setShouldReplace] = React.useState(false);
+  const [
+    leaderboardsToReplace,
+    setLeaderboardsToReplace,
+  ] = React.useState<?Array<string>>(null);
+
+  const authenticatedUser = React.useContext(AuthenticatedUserContext);
+
+  const [erroredLeaderboards, setErroredLeaderboards] = React.useState<
+    Array<ErroredLeaderboard>
+  >([]);
+
+  const [progress, setProgress] = React.useState(0);
+  const [onRetry, setOnRetry] = React.useState<?RetryOrAbandonCallback>(null);
+  const [onAbandon, setOnAbandon] = React.useState<?RetryOrAbandonCallback>(
+    null
+  );
+
+  /**
+   * Launch the replacement of leaderboards, if needed.
+   * First, register game, then copy leaderboards in database, and finally
+   * replace them in events.
+   */
+  const ensureLeaderboardsAreReplaced = React.useCallback(
+    async () => {
+      const { getAuthorizationHeader, profile } = authenticatedUser;
+      if (!leaderboardsToReplace || !project || !gameId) {
+        throw new Error('No leaderboards fetched from events sheet.');
+      }
+      if (!profile) {
+        throw new Error('User is not connected.');
+      }
+
+      // Number of leaderboards to replace + game registration + leaderboard replace in events sheet.
+      const totalSteps = leaderboardsToReplace.length + 2;
+      const progressStep = (1 / totalSteps) * 100;
+
+      const replacedLeaderboardsMap = {};
+
+      setProgress(0);
+      setOnRetry(null);
+      setOnAbandon(null);
+      setErroredLeaderboards([]);
+
+      // Register game.
+      try {
+        await registerGame(getAuthorizationHeader, profile.id, {
```

Comment:
- If this errors, the user can still retry right? Because leaderboards will fail and it will a retry button?
- And if the game exists, it's silently caught? 

=> comment about these two cases in the catch clause